### PR TITLE
feat: fixes from hba into core

### PR DESCRIPTION
**Pull request by YazeedLoonat**
This brings across the following changes from HBA -> core

https://github.com/housingbayarea/bloom/pull/667 :- disallowing of malicious first/last name on public user creation

https://github.com/housingbayarea/bloom/pull/669 :- AMI chart fixes and password outdated error messaging

https://github.com/housingbayarea/bloom/pull/670 :- limiting public site closed listings

https://github.com/housingbayarea/bloom/pull/671 :- new "mostRecentlyCreated" applicant endpoint, update so forgot password marks a user as confirmed

https://github.com/housingbayarea/bloom/pull/672 :- update to externalizedListing endpoint for doorway ingestion

https://github.com/housingbayarea/bloom/pull/674 :- application security patch

### CODE REVIEW

**Comment by emilyjablonski** [comment]
Can you add a description to this PR? If anything just for documentation

**Comment by YazeedLoonat** [comment]
Hey @emilyjablonski added some context!

File: api/test/integration/application.e2e-spec.ts [hunk 1]
```diff
@@ -652,6 +655,7 @@ describe('Application Controller Tests', () => {
       let geocodingOptions = savedPreferences[0].options[0];
       // This catches the edge case where the geocoding hasn't completed yet
       if (geocodingOptions.extraData.length === 1) {
+        // I'm unsure why removing this console log makes this test fail. This should be looked into
```
Comment (ludtkemorgan) [inline api/test/integration/application.e2e-spec.ts:658]: I wrote this comment, and I'm still unsure about it

**Review by ludtkemorgan — APPROVED** [review]

**Comment by ludtkemorgan** [comment]
Looks like the test coverage threshold dropped below 90%
